### PR TITLE
Allow a user to select text in message boxes.

### PR DESCRIPTION
**Pull request by dbrnz**
This makes it easy to copy text in message boxes...

### CODE REVIEW

**Comment by agarny** [comment]
Hmm, your change is breaking things. The reason I had `messagebox-text-interaction-flags` set to `4` was to prevent the text of any dialog box from being selected (among other things; as mentioned in the comment just above that line). So, to set it to `5` is not an option.

I guess you could set `messagebox-text-interaction-flags` to `5` for the dialog box for which you would like the text to be selectable?...

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -292,8 +292,8 @@ showEnableAction(mGui->actionPreferences, false);
     // don't want to allow at all
 
     qApp->setStyleSheet("QMessageBox {"
-                        "    messagebox-text-interaction-flags: 4;"
-                        "}");
+                        "    messagebox-text-interaction-flags: 5;" // Mouse clickable links, mouse selectable text
+                        "}");                                       // Or 13 to also allow keyboard clickable links
```
Comment (agarny) [inline src/mainwindow.cpp]: FYI: your comment doesn't respect the style used in OpenCOR.

**Comment by dbrnz** [comment]
I wondered about your reasons hence thought it best for you to review the PR. Yes, I can set the appropriate flags for my specific message boxes, but I'm curious as to why doing so in general breaks OpenCOR. Wouldn't it be more user friendly to allow copy/paste from message boxes (e.g. when seeking help about error messages) and explicitly disallow selection for just those message boxes which break other functionality?

**Comment by agarny** [comment]
It's just that on Windows, I have always been used to dialog boxes with non selectable text and this is therefore the behaviour I decided to reproduce by setting `messagebox-text-interaction-flags ` to `4`. To me, this looks better for things like OpenCOR's About box.

Now, I do appreciate that there are cases where it would indeed be useful to copy/paste from a dialog box.

Ok, let's forget this PR. I am going to remove altogether the line of code that you wanted to modify and I am going to create a class that inherits from `QMessageBox` and that allows the developer to specify which options it wants / doesn't want. I can do that before the end of today, if you can wait a bit.

**Comment by dbrnz** [comment]
I'm in no real rush for this so whenever best suits you (it was prompted by the comment @nickerso made in issue #1066, which I've resolved with clickable links).

Presumably you are aware of [Qt::TextInteractionFlags](http://doc.qt.io/qt-5/qt.html#TextInteractionFlag-enum) that can be set on a QMessageBox??

**Comment by agarny** [comment]
> Presumably you are aware of [Qt::TextInteractionFlags](http://doc.qt.io/qt-5/qt.html#TextInteractionFlag-enum) that can be set on a `QMessageBox`??

Yes, I am aware of it. It's just that setting the style of the application was an easy of getting what I was after. Anyway, will see if I can get it done today.

**Comment by agarny** [comment]
For information, a 'default' call to, say, `QMessageBox::warning()` will result in the text being selectable on OS X, but not on Windows...

**Comment by agarny** [comment]
... so, rather than using `QMessageBox::warning()` and the like, we should use the soon to come `warningMessageBox()` and the like (see issue #1077) since those methods ask for some text interaction flags. In other words, whenever we will want to show a/n information/question/warning/critical/about message box, we will have to think about the text interaction flags that we want to use (thus making the message box truly cross-platform).

File: src/mainwindow.cpp (hunk 1 above)
Comment (dbrnz) [inline src/mainwindow.cpp]: Old habits die hard... :)

Comment (agarny) [inline src/mainwindow.cpp]: No worries, I shall keep reminding you about the OpenCOR style whenever needed. :)

**Comment by agarny** [comment]
FWIW, I thought it through a bit more and I am not sure we need the developer to specify the text interaction flags. Indeed, depending on the type of message box (information, question, warning, critical or about), the text interaction flags should always be the same (for consistency throughout OpenCOR, if anything).

**Comment by agarny** [comment]
This PR is not relevant anymore (see issue #1077), so closing.

**Pull request closed without merging**